client/me/account/account-email-field.tsx

```diff
@@ -209,6 +187,37 @@ const AccountEmailField = ( {
 		dispatch( setUserSetting( 'user_email', value ) );
 	};
 
+	const scrollAndFocus = useCallback( () => {
+		if ( isEmailChangePending ) {
+			unlockRef.current?.focus();
+		} else {
+			inputRef.current?.focus();
+		}
+		// We need to bump the scroll call to the back of the callstack, since the dialog that
+		// triggers this on close can influence the scroll window.
+		setTimeout( () => {
+			inputRef.current?.scrollIntoView( { behavior: 'smooth', block: 'start' } );
+		} );
+	}, [ isEmailChangePending ] );
+
+	useEffect( () => {
+		// Check for query param used to indicate a need to scroll to the input, this is added when
+		// redirecting to this page for the purpose of email change.
+		const urlQueryParams = new URL( window.location.href )?.searchParams;
+		const focusEmail = urlQueryParams.get( 'focusEmail' );
+		if ( focusEmail ) {
+			scrollAndFocus();
+			window.history.replaceState( {}, '', removeQueryArgs( window.location.href, 'focusEmail' ) );
+		}
+		// Listen for an event signalling to focus and scroll to the email field or button. This
+		// happens when we are already on a page with the input and need to trigger the
+		// functionality.
+		emailFormEventEmitter.addEventListener( 'highlightInput', scrollAndFocus );
```

Comment:
Instead of using a query param for cases coming from another page, and an event emitter for cases on triggering this on the same page, I think the other option is moving this to Redux but I am conflicted. In both cases this is more of a signal/event to trigger functionality and not state to manage, so Redux seems a little weird but we could probably manage it with a true/false toggle. But that seems like potential to get out-of-sync in some way (?), and generally it feels like extra boilerplate code and overkill for the purpose. 

Id be curious thoughts of other developers reviewing.

Reply:
I think the query param implementation makes sense. But does it need the same `isEmailChangePending` conditional check?

Also, I think the scroll only works the first time in Firefox. 🤔 

https://github.com/user-attachments/assets/4f7c0072-825a-413b-a94b-620a5fc37119

Reply:
> But does it need the same isEmailChangePending conditional check?

Yeah. If the change is pending the input is disabled, so instead we focus the button that cancels the pending change and unlocks the input. So either way that check is relevant to figure out what element to focus.

Reply:
> Also, I think the scroll only works the first time in Firefox. 🤔

Ah interesting! I don't think its just firefox.

So if you open the modal on `/me/account` and click "change" scroll works every time. But if you go to `/me`, open the modal, and click "change", the scroll/focus it only works if you haven't loaded /me/account yet.

I didn't think this would be the case but for whatever reason it seems like the email form component on /me/account doesn't dismount when you go back to /me. So the hook checking for the query param on mount to scroll to the input isn't firing since the component has been mounted and none of its dependencies have changed.

Reply:
Ok not component mounting like i suspected, its the component loading before the window url is set in this case.

this should now be resolved with [df256f2](https://github.com/Automattic/wp-calypso/pull/94683/commits/df256f250eeda64b11023d73c04e7a267ecd11be).